src/cloudant/client.py

```diff
@@ -290,6 +290,8 @@ def create_database(self, dbname, partitioned=False, **kwargs):
         except CloudantDatabaseException as ex:
             if ex.status_code == 412:
                 raise CloudantClientException(412, dbname)
+            if ex.status_code == 400:
```

Comment:
I don't think we should special case `400` here, I think the problem is that `412` is special cased without an `else` suppressing all the other exceptions. For example if the service returns a `401` or a `500` we should still be raising here. So I think what we really want is:
```python
else:
  raise ex
```

Reply:
Oops, I missed this earlier. Didn't refresh the page. I'm agreement; we need to catch all failures here.

Reply:
We have custom messages for `CloudantDatabaseException` and `CloudantClientException` in `_messages.py`.  If we want to catch and throw 500 under a database exception I'll add the message `The server encountered an unexpected condition which prevented it from fulfilling the request.` to the messages module.